builtin/providers/azurerm/resource_arm_dns_cname_record.go

```diff
@@ -34,10 +34,16 @@ func resourceArmDnsCNameRecord() *schema.Resource {
 			},
 
 			"records": &schema.Schema{
-				Type:     schema.TypeSet,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Elem:       &schema.Schema{Type: schema.TypeString},
+				Set:        schema.HashString,
+				Deprecated: "Use `record` instead. This attribute will be removed in a future version",
```

Comment:
I believe we have `Removed` which is an error instead of a warning. I think that's what you want here since we don't handle "records" anymore right?